// react-ions/lib/styles/common/screen.css.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
// Minima of each width category
var viewportWidthTiny = 375;
var viewportWidthSmall = 768;
var viewportWidthMedium = 992;
var viewportWidthLarge = 1200;
var viewportWidthXLarge = 1680;
var viewportHeightSmall = 600;

// Maxima of each width category
var viewportWidthExtraSmallMax = viewportWidthSmall - 1;
var viewportWidthSmallMax = viewportWidthMedium - 1;
var viewportWidthMediumMax = viewportWidthLarge - 1;
var viewportWidthLargeMax = viewportWidthXLarge - 1;
var viewportHeightSmallMax = viewportHeightSmall - 1;

// Strings usable directly after '@media'
var mediaSmallMinus = exports.mediaSmallMinus = "@media(max-width: " + viewportWidthSmallMax + "px)";
var mediaSmallPlus = exports.mediaSmallPlus = "@media(min-width: " + viewportWidthSmall + "px)";
var mediaMediumPlus = exports.mediaMediumPlus = "@media(min-width: " + viewportWidthMedium + "px)";
var mediaMediumMinus = exports.mediaMediumMinus = "@media(max-width: " + viewportWidthMediumMax + "px)";
var mediaLargePlus = exports.mediaLargePlus = "@media(min-width: " + viewportWidthLarge + "px)";
var mediaLargeMinus = exports.mediaLargeMinus = "@media(max-width: " + viewportWidthLargeMax + "px)";

var mediaTiny = exports.mediaTiny = "@media(max-width: " + viewportWidthTiny + "px)";
var mediaExtraSmall = exports.mediaExtraSmall = "@media(max-width: " + viewportWidthExtraSmallMax + "px)";
var mediaSmall = exports.mediaSmall = "@media(" + mediaSmallPlus + " and " + mediaSmallMinus;
var mediaMedium = exports.mediaMedium = "@media(" + mediaMediumPlus + " and " + mediaMediumMinus;
var mediaLarge = exports.mediaLarge = "@media(" + mediaLargePlus + " and " + mediaLargeMinus;
var mediaXLarge = exports.mediaXLarge = "@media(min-width: " + viewportWidthXLarge + "px)";

var mediaHeightSmall = exports.mediaHeightSmall = "@media(max-height: " + viewportHeightSmallMax + "px)";
