// react-ions/lib/components/Chip/Chip.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _OptClass = require('../internal/OptClass');

var _OptClass2 = _interopRequireDefault(_OptClass);

var _style = require('./style.scss');

var _style2 = _interopRequireDefault(_style);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Chip = function Chip(props) {
  var isActionable = props.clickCallback ? 'is-actionable' : null;
  var chipClasses = (0, _OptClass2.default)(_style2.default, ['chip-wrapper', props.color, props.size, props.optClass, props.className, isActionable]);

  return _react2.default.createElement(
    'span',
    { className: chipClasses, style: props.style },
    _react2.default.createElement(
      'span',
      { onClick: props.clickCallback },
      props.text
    )
  );
};

Chip.propTypes = {
  /**
   * The background color of the chip (see Foundations > Colors for more info).
   */
  color: _propTypes2.default.oneOf(['primary', 'primary-1', 'primary-2', 'primary-3', 'primary-4', 'primary-5', 'primary-darker', 'primary-6', 'neutral-1', 'neutral-2', 'neutral-3', 'neutral-4', 'danger', 'success', 'warning', 'info']),
  /**
   * Optional click callback.
   */
  clickCallback: _propTypes2.default.func,
  /**
   * Optional styles to add to the chip.
   */
  optClass: _propTypes2.default.string,
  /**
   * Optional size of the chip.
   */
  size: _propTypes2.default.oneOf(['larger', 'smaller']),
  /**
   * Optional style object to be added for additional CSS.
   */
  style: _propTypes2.default.object,
  /**
   * Text to display inside the chip.
   */
  text: _propTypes2.default.string
};

Chip.defaultProps = {
  color: 'primary'
};

exports.default = Chip;
